src/ncp/ncp_spi.cpp

```diff
@@ -102,132 +102,136 @@ static uint16_t spi_header_get_data_len(const uint8_t *header)
+
+    otPlatSpiSlavePrepareTransaction(mEmptySendFrameZeroAccept, kSpiHeaderLength, mEmptyReceiveFrame, kSpiHeaderLength,
+                                     true);
 }
 
-void NcpSpi::SpiTransactionComplete(void *aContext, uint8_t *anOutputBuf, uint16_t aOutputBufLen, uint8_t *aInputBuf,
+void NcpSpi::SpiTransactionComplete(void *aContext, uint8_t *aOutputBuf, uint16_t aOutputBufLen, uint8_t *aInputBuf,
                                     uint16_t aInputBufLen, uint16_t aTransactionLength)
 {
-    static_cast<NcpSpi*>(aContext)->SpiTransactionComplete(anOutputBuf, aOutputBufLen, aInputBuf, aInputBufLen,
+    static_cast<NcpSpi*>(aContext)->SpiTransactionComplete(aOutputBuf, aOutputBufLen, aInputBuf, aInputBufLen,
                                                            aTransactionLength);
 }
 
-void NcpSpi::SpiTransactionComplete(uint8_t *aMISOBuf, uint16_t aMISOBufLen, uint8_t *aMOSIBuf, uint16_t aMOSIBufLen,
-                                    uint16_t aTransactionLength)
+void NcpSpi::SpiTransactionComplete(uint8_t *aOutputBuf, uint16_t aOutputBufLen, uint8_t *aInputBuf,
+                                    uint16_t aInputBufLen, uint16_t aTransactionLength)
 {
     // This may be executed from an interrupt context.
     // Must return as quickly as possible.
 
-    uint16_t rx_data_len(0);
-    uint16_t rx_accept_len(0);
-    uint16_t tx_data_len(0);
-    uint16_t tx_accept_len(0);
+    uint16_t rx_data_len = 0;
+    uint16_t rx_accept_len = 0;
+    uint16_t tx_data_len = 0;
+    uint16_t tx_accept_len = 0;
 
     // TODO: Check `PATTERN` bits of `HDR` and ignore frame if not set.
     //       Holding off on implementing this so as to not cause immediate
     //       compatibility problems, even though it is required by the spec.
 
     if (aTransactionLength >= kSpiHeaderLength)
     {
-        if (aMISOBufLen >= kSpiHeaderLength)
+        if (aOutputBufLen >= kSpiHeaderLength)
         {
-            rx_accept_len = spi_header_get_accept_len(aMISOBuf);
-            tx_data_len = spi_header_get_data_len(aMISOBuf);
-            (void)spi_header_get_flag_byte(aMISOBuf);
+            rx_accept_len = spi_header_get_accept_len(aOutputBuf);
+            tx_data_len = spi_header_get_data_len(aOutputBuf);
+            (void)spi_header_get_flag_byte(aOutputBuf);
         }
 
-        if (aMOSIBufLen >= kSpiHeaderLength)
+        if (aInputBufLen >= kSpiHeaderLength)
         {
-            rx_data_len = spi_header_get_data_len(aMOSIBuf);
-            tx_accept_len = spi_header_get_accept_len(aMOSIBuf);
+            rx_data_len = spi_header_get_data_len(aInputBuf);
+            tx_accept_len = spi_header_get_accept_len(aInputBuf);
         }
 
-        if ( !mHandlingRxFrame
-          && (rx_data_len > 0)
-          && (rx_data_len <= (aTransactionLength - kSpiHeaderLength))
-          && (rx_data_len <= rx_accept_len)
-        ) {
+        if (!mHandlingRxFrame &&
```

Comment:
Just venting, the following isn't necessarily a criticism of this pull request:

I hate having operators on the right because makes commenting out individual tests when debugging more error-prone. But if that is what we are using for our style guide now, then I suppose we are better off with consistency.